content/en/docs/develop/porting.md

```diff
@@ -614,23 +614,333 @@ To make a patch:
+
+As best practice, we recommend to begin the file with defining dependencies with an invisible boolean option that is set to `y` (i.e. an option that will not be visible in the `menuconfig` screen):
+
+```kconfig
+### Invisible option for dependencies
+config APPHELLOWORLD_DEPENDENCIES
+	bool
+	default y
+	# dependencies with `select`:
+	select LIBNOLIBC if !HAVE_LIBC
+	select LIB1
+	select LIB2
+```
+
+In this example, `LIB1` and `LIB2` would be enabled (the user can't unselect them).
+Additionally, if the user did not provide and select any `libc`, the Unikraft internal replacement `nolibc` would be selected.
+Of course, you could also directly define a dependency on a particular `libc` (e.g., `libmusl`), instead.
+You can also depend on feature flags (like `HAVE_LIBC`) to provide or hide options.
+The feature flag `HAVE_LIBC` in this example is set as soon as a proper and full-fledged `libc` was selected by the user.
+You can get an overview of available feature flags in `libs/Config.uk`.
+
+Any other setting of your application can be a type of boolean, int, or string.
+You are even able to define dropdown-selections.
+You can find a documentation of the syntax at the [Linux kernel tree](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/plain/Documentation/kbuild/kconfig-language.txt).
+Please note that Unikraft does not have tristates (the `m` option, for kernel modules, is not available).
+
+```kconfig
+  ### App configuration
+  config APPHELLOWORLD_OPTION
+  	bool "Boolean Option 1"
+  	default y
+  	help
+  	  Help text of Option 1
+```
+
+### exportsyms.uk
+
+Unikraft provides separate namespaces for each library.
+This means that every function and variable will only be visible and linkable internally.
+
+To make a symbol visible for other libraries, add it to this `exportsyms.uk` file.
+It is simply a flat file, with one symbol name per line.
+Line comments may be introduced by the hash character (`#`).
+This option may be given more than once.
+
+If you are writing an application, you need to add your program entry point to this file (this is `main` if you use the `ukboot` internal library).
+Most likely nothing else should be there.
+For a library, all external API functions must be listed.
+
+For the sake of file structure consistency, it is not recommended to change the default path of this symbols file, unless it is really necessary (e.g., multiple libraries are sharing the same base folder, this symbols file is part of a remotely fetched archive).
+You can override it by defining the `APPNAME_EXPORTS` variable.
+The path must be either absolute (you can refer with `$(APPNAME_BASE)` to the base directory of your application sources) or relative to the Unikraft sources directory.
+
+**Note: If no `exportsyms.uk` file is present within a given library (internal or external), all the symbols will be exported by default.**
```

Comment:
I would use the [`alert` shortcode](https://www.docsy.dev/docs/adding-content/shortcodes/#alert) for notes.